demo_path_routing/views.py

```diff
@@ -0,0 +1,92 @@
+import os
+import re
+
+from django import forms
+from django.conf import settings
+from django.http import Http404, HttpResponse, HttpResponseRedirect
+from django.shortcuts import render
+
+from django_docker_engine.docker_utils import (DockerClientRunWrapper,
+                                               DockerClientSpec,
+                                               DockerContainerSpec)
+
+from .forms import LaunchForm, UploadForm
+from .tools import tools
+from .utils import hostname
+
+client = DockerClientRunWrapper(
+    DockerClientSpec(None, do_input_json_envvar=True))
+UPLOAD_DIR = os.path.join(os.path.dirname(__file__), 'upload')
+
+
+def index(request):
+    launch_form = LaunchForm()
+    # TODO: Pass this info through the constructor
+    launch_form.fields['data'] = forms.ChoiceField(
+        widget=forms.Select,
+        choices=((f, f) for f in os.listdir(UPLOAD_DIR) if f != '.gitignore')
+    )
+    launch_form.initial['data'] = request.GET.get('uploaded')
+
+    context = {
+        'container_names': [container.name for container in client.list()],
+        'launch_form': launch_form,
+        'upload_form': UploadForm(),
+    }
+    return render(request, 'index.html', context)
+
+
+def launch(request):
+    if request.method == 'POST':
+        form = LaunchForm(request.POST)
+        if form.is_valid():
+            post = form.cleaned_data
+
+            input_url = 'http://{}:{}/upload/{}'.format(
+                hostname(), request.get_port(), post['data'])
+            tool_spec = tools[post['tool']]
+
+            container_name = post['container_name']
+            container_path = '/docker/{}/'.format(container_name)
+            container_spec = DockerContainerSpec(
+                container_name=container_name,
+                image_name=tool_spec['image'],
+                input=tool_spec['input'](input_url, container_path))
+            client.run(container_spec)
+            return HttpResponseRedirect(container_path)
+
+
+def kill(request, name):
+    if request.method == 'POST':
+        container = client.list(filters={'name': name})[0]
+        container.remove(
```

Comment:
Could we reuse the `client.purge()` here?

Reply:
Killing single containers by name wasn't something Refinery needed, so I hadn't put it in our wrapper... perhaps it should be added there, rather than digging into the internals here?